docs/concepts/style_guide.md

```diff
@@ -0,0 +1,178 @@
+# Beaker Style Guide
+
+## Scope of this guide
+
+The purpose of this guide is to provide definitions for preferred practices when
+writing Beaker tests, both syntactically and stylistically. This guide will
+define and provide examples for preferred test layout and conventions. Common
+patterns that are recommend as well as patterns that should be avoided will be
+described.
+
+No style manual can cover every possible circumstance. When a judgement call
+becomes necessary, keep in mind the following general ideas:
+
+1. **Readability matters**. If you have to choose between two equally effective
+alternatives, pick the more readable one. This is, of course, subjective, but if
+you can read your own code three months from now, that’s a great start. Don’t be
+clever over readable, unless you have a documented purpose. Use object oriented
+programming when things get complex and DRY.
+2. **Inherit upstream conventions**. Beaker is still ruby, so use the
+[ruby community style guide](https://github.com/bbatsov/ruby-style-guide). When
+not called out here, use the ruby style guide.
+
+## Test Naming
+
+Tests should test what they say they test. Test names, both the name of the test
+file and the value given to the
+[`test_name`](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Structure#test_name-instance_method)
+function should provide an accurate indication about the purpose of the test.
+
+The `test_name` function should be the first line in the Beaker test file.
+
+**Good:**
+```ruby
+# head -1 puppet/acceptance/tests/resource/file/should_default_mode.rb
+test_name "file resource: set default modes"
```

Comment:
Since we are advocating for structure methods to use blocks, the good example should include a `do` block opener.